automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala

```diff
@@ -351,5 +353,116 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
+          }
+          actualBucketRolesAndPolicies should contain theSameElementsAs expectedBucketRolesAndPolicies
+
+          // check object acls
+          val unconstrainedObjectAcl = googleStorageDAO.getDefaultObjectAccessControls(bucketName).futureValue.getItems.asScala.toList
+          // service account should have owner access
+          assert(unconstrainedObjectAcl.exists(acl => acl.getRole().equals("OWNER") && acl.getEmail().endsWith(serviceAccountEmailDomain)))
+          val actualObjectRolesAndPolicies = unconstrainedObjectAcl.collect {
+            case acl if (acl.getRole() != null && !acl.getRole().equals("OWNER")) => (acl.getRole(), acl.getEmail())
+          }
+          val expectedObjectRolesAndPolicies = unconstrainedSamPolicies.collect {
+            case AccessPolicyResponseEntry("project-owner", AccessPolicyMembership(emails, _, _), _) => ("READER", emails.head)
+            case AccessPolicyResponseEntry(policyName, _, email) if policyToObjectAccessLevel.contains(policyName) => (policyToObjectAccessLevel(policyName), email.value)
+          }
+          actualObjectRolesAndPolicies should contain theSameElementsAs expectedObjectRolesAndPolicies
+        }
+      }
+    }
+
+    "should have correct policies in Sam and ACLs in Google when a constrained workspace is created" in {
+      implicit val patienceConfig: PatienceConfig = PatienceConfig(timeout = 20 seconds)
+      implicit val token: AuthToken = ownerAuthToken
+      val workspacePolicyNames = Set("can-compute", "project-owner", "writer", "reader", "share-writer", "can-catalog", "owner", "share-reader")
+      // bucket and object access levels for sam policies as described in comments in insertBucket function in HttpGoogleServicesDAO
+      val policyToBucketAccessLevel = Map("project-owner" -> "WRITER", "owner" -> "WRITER", "writer" -> "WRITER", "reader" -> "READER")
+      val policyToObjectAccessLevel = Map("project-owner" -> "READER", "owner" -> "READER", "writer" -> "READER", "reader" -> "READER")
+      val serviceAccountEmailDomain = "developer.gserviceaccount.com"
+
+      withCleanBillingProject(owner) { projectName =>
+        withGroup("authDomain", List(owner.email)) { authDomain =>
+          withWorkspace(projectName, s"constrained-workspace", Set(authDomain)) { workspaceName =>
+            val workspaces = Rawls.workspaces.list()
+            val workspaceId = workspaces.parseJson.asInstanceOf[JsArray].elements.flatMap { workspaceListEntry =>
+              workspaceListEntry.asJsObject.getFields("workspace").collect {
+                case workspace if (workspace.asJsObject.getFields("name").exists(name => name.convertTo[String].equals(workspaceName)) &&
+                  workspace.asJsObject.getFields("namespace").exists(namespace => namespace.convertTo[String].equals(projectName))) => workspace.asJsObject.getFields("workspaceId").head
+              }
+            }.head.convertTo[String]
```

Comment:
There are some repeated blocks of code in these two tests (like parsing out the workspace Ids) that you could pull out into a separate function